components/brave_rewards/browser/database/publisher_info_database.cc

```diff
@@ -1132,6 +1129,57 @@ ledger::ServerPublisherInfoPtr PublisherInfoDatabase::GetServerPublisherInfo(
 
 // Other -------------------------------------------------------------------
 
+void PublisherInfoDatabase::RecordP3AStats(bool auto_contributions_on) {
```

Comment:
Please move this into a separate file under `database` folder. And just proxy it through this file. New file could maybe be `database_p3a.h/cc`?

Reply:
@NejcZdovc I think we've already discussed this during the first pass over rewards P3A :) I still suggest to keep recording statistics close to the business logic - this is a common pattern in Chromium (you can check literally any DB, e.g. `login_database.h`)